src/System.Private.CoreLib/shared/System/CustomAttribute.cs

```diff
@@ -0,0 +1,472 @@
+
+            if (ArgumentType.IsEnum)
+                return string.Format(typed ? "{0}" : "({1}){0}", Value, ArgumentType.FullName);
+
+            else if (Value == null)
+                return string.Format(typed ? "null" : "({0})null", ArgumentType.Name);
+
+            else if (ArgumentType == typeof(string))
+                return string.Format("\"{0}\"", Value);
+
+            else if (ArgumentType == typeof(char))
+                return string.Format("'{0}'", Value);
+
+            else if (ArgumentType == typeof(Type))
+                return string.Format("typeof({0})", ((Type)Value).FullName);
+
+            else if (ArgumentType.IsArray)
+            {
+                IList<CustomAttributeTypedArgument> array = (IList<CustomAttributeTypedArgument>)Value;
+
+                Type elementType = ArgumentType.GetElementType();
+                string result = string.Format(@"new {0}[{1}] {{ ", elementType.IsEnum ? elementType.FullName : elementType.Name, array.Count);
+
+                for (int i = 0; i < array.Count; i++)
+                {
+                    result += string.Format(i == 0 ? "{0}" : ", {0}", array[i].ToString(elementType != typeof(object)));
+                }
+
+                result += " }";
+
+                return result;
+            }
+
+            return string.Format(typed ? "{0}" : "({1}){0}", Value, ArgumentType.Name);
+        }
+
+        public override int GetHashCode() => base.GetHashCode();
+        public override bool Equals(object obj) => obj == (object)this;
+
+        public Type ArgumentType => m_argumentType;
+        public object Value => m_value;
+    }
+
+    internal enum CustomAttributeEncoding : int
+    {
+        Undefined = 0,
+        Boolean = CorElementType.ELEMENT_TYPE_BOOLEAN,
+        Char = CorElementType.ELEMENT_TYPE_CHAR,
+        SByte = CorElementType.ELEMENT_TYPE_I1,
+        Byte = CorElementType.ELEMENT_TYPE_U1,
+        Int16 = CorElementType.ELEMENT_TYPE_I2,
+        UInt16 = CorElementType.ELEMENT_TYPE_U2,
+        Int32 = CorElementType.ELEMENT_TYPE_I4,
+        UInt32 = CorElementType.ELEMENT_TYPE_U4,
+        Int64 = CorElementType.ELEMENT_TYPE_I8,
+        UInt64 = CorElementType.ELEMENT_TYPE_U8,
+        Float = CorElementType.ELEMENT_TYPE_R4,
+        Double = CorElementType.ELEMENT_TYPE_R8,
+        String = CorElementType.ELEMENT_TYPE_STRING,
+        Array = CorElementType.ELEMENT_TYPE_SZARRAY,
+        Type = 0x50,
+        Object = 0x51,
+        Field = 0x53,
+        Property = 0x54,
+        Enum = 0x55
+    }
+
+    [StructLayout(LayoutKind.Auto)]
+    internal readonly partial struct CustomAttributeEncodedArgument
+    {
+        private readonly long m_primitiveValue;
+        private readonly CustomAttributeEncodedArgument[] m_arrayValue;
+        private readonly string m_stringValue;
+        private readonly CustomAttributeType m_type;
+
+        internal static void ParseAttributeArguments(ConstArray attributeBlob,
```

Comment:
Where is this called from?

Reply:
this type is not needed anymore in this PR